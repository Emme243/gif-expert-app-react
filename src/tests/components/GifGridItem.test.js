import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Pruebas en GifGridItem.js component', () => {
  const title = 'Un título';
  const url = 'https://localhots/ld.png';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de tener un párrafo con el title', () => {
    const pText = wrapper.find('p').text().trim();
    expect(pText).toBe(title);
  });

  test('debe de tener la imagen con el url y alt de los props', () => {
    const imgAttrs = wrapper.find('img').props();
    // console.log(imgAttrs); // {src: 'Algo', alt: 'Algo'}
    expect(imgAttrs.src).toBe(url);
    expect(imgAttrs.alt).toBe(title);
  });

  test('debe de tener animate__fadeIn', () => {
    const div = wrapper.find('div');
    expect(div.prop('className')).toContain('animate__fadeIn');
  });
});
