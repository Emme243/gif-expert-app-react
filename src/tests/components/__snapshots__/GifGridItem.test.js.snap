// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem.js component should match snapshot 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un título"
    src="https://localhots/ld.png"
  />
  <p>
    Un título
  </p>
</div>
`;
